src/function/vector_cast_functions.cpp

```diff
@@ -34,13 +34,13 @@ struct CastChildFunctionExecutor {
 
 static void resolveNestedVector(std::shared_ptr<ValueVector> inputVector, ValueVector* resultVector,
     uint64_t numOfEntries, CastFunctionBindData* dataPtr) {
-    auto inputType = inputVector->dataType;
-    auto resultType = resultVector->dataType;
+    const auto* inputType = &inputVector->dataType;
```

Comment:
Why `const auto*` ? You seem to use `const auto&` everywhere.

Reply:
inputType has to be able to point to different addresses in this case